### ISSUE TITLE
Reproducing GRN-2B GenEval score: 0.715 vs reported 0.76

### ISSUE THREAD

**Issue by wangtong627** [bytedance/GRN#7]
Hi, thanks for releasing GRN and the checkpoints.

I am trying to reproduce the GRN-2B Text-to-Image GenEval result reported in Table 4 of the paper. The paper reports:

| Model | Single Obj. | Two Obj. | Count | Colors | Pos. | Color Attr. | Overall |
|---|---:|---:|---:|---:|---:|---:|---:|
| GRN 2B | 0.99 | 0.90 | 0.72 | 0.84 | 0.52 | 0.60 | 0.76 |

With my current evaluation setup, I get a noticeably lower overall score, mainly due to the `position` category:

| Metric | My result | Paper |
|---|---:|---:|
| Single Obj. | 0.9938 | 0.99 |
| Two Obj. | 0.8662 | 0.90 |
| Count | 0.7250 | 0.72 |
| Colors | 0.8245 | 0.84 |
| Position | 0.3250 | 0.52 |
| Color Attr. | 0.5550 | 0.60 |
| Overall | 0.7149 | 0.76 |

My GenEval summary is:

```text
Total images: 2212
Total prompts: 553
% correct images: 70.30%
% correct prompts: 85.53%

two_object       = 86.62% (343 / 396)
color_attr       = 55.50% (222 / 400)
counting         = 72.50% (232 / 320)
position         = 32.50% (130 / 400)
single_object    = 99.38% (318 / 320)
colors           = 82.45% (310 / 376)

Overall score (avg. over tasks): 0.71490
```

My generation/evaluation config:

```text
Base repo commit: 62c94fb187d129cfc007e3804106d1336f6cf414

Model: GRN2b
Task: T2I
PN: 1M
Checkpoint: GRN_T2I_2B_FSA_137200.pth
VAE: HBQ_image_video_tokenizer_64dim_M4_20260626.ckpt
Text encoder: umt5-xxl

GenEval metadata: evaluation_metadata.jsonl, 553 prompts
Samples per prompt: 4
Seed rule: base seed 42, sample seed = 42 + prompt_idx * 4 + sample_idx
Prompt format: "<T2I>" prefix added to each GenEval prompt
Prompt rewrite: disabled

guidance_scale: 3.0
CFG enabled flag: cfg = 1
CFG value: cfg_val = 3.0
CFG type: cfg_interval_0.0
CFG insertion layer: 0
CFG list logged during inference: [3.0]
Negative prompt: ""
temperature: 1.1
max_steps: 50
h_div_w: 1.0
snr_shift: 1.0
complexity_aware_Tmin: 10
complexity_aware_Tmax: 50
complexity_aware_k: 0
complexity_aware_b: 50
complexity_aware_wp: 5
```

For detection/summarization I used the standard GenEval-style pipeline:

```text
Mask2Former config:
mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco.py

Mask2Former weights:
mask2former_swin-s-p4-w7-224_lsj_8x2_50e_coco.pth

det.jsonl lines: 2212
```

I could not find an official script/command for reproducing the T2I GenEval result in the README, so I may be missing some evaluation detail.

Could you please clarify:

1. What exact command/config was used to obtain the Table 4 GenEval score for GRN-2B?
2. Which checkpoint should be used for that number? The README links `GRN_T2I_2B.pth`, while my local checkpoint is named `GRN_T2I_2B_FSA_137200.pth`.
3. Was prompt rewriting, a different prompt prefix, or any special negative prompt used?
4. Was the reported score from a single seed or averaged over multiple seeds?
5. Which GenEval version / metadata file / Mask2Former config and weights were used?
6. Are there any known settings that especially affect the `position` category?

Thanks!

**Comment by HelloJianHan** [comment]
@wangtong627 The main difference is prompt rewriting. I use the rewrited prompts from [here](https://github.com/baaivision/URSA/blob/main/evaluations/geneval/prompts.json). GRN_T2I_2B_FSA_137200.pth is the newest checkpoint.

**Comment by HelloJianHan** [comment]
In fact, GenEval has been pushed above 95% by numerous methods. The metrics of this benchmark are heavily dependent on the training prompt distribution, training data, and whether reinforcement learning is adopted. For this reason, I actually recommend evaluating model performance via side-by-side human comparisons instead.
